arch/xtensa/src/esp32/esp32_spiflash.c

```diff
@@ -1044,24 +1427,12 @@ static ssize_t esp32_read(FAR struct mtd_dev_s *dev, off_t offset,
                           size_t nbytes, FAR uint8_t *buffer)
 {
   int ret;
-  uint8_t *tmpbuff = buffer;
   FAR struct esp32_spiflash_s *priv = MTD2PRIV(dev);
 
 #ifdef CONFIG_ESP32_SPIFLASH_DEBUG
   finfo("esp32_read(%p, 0x%x, %d, %p)\n", dev, offset, nbytes, buffer);
 #endif
 
-#ifdef CONFIG_XTENSA_USE_SEPARATE_IMEM
-  if (esp32_ptr_extram(buffer))
-    {
-      tmpbuff = xtensa_imm_malloc(nbytes);
-      if (tmpbuff == NULL)
-        {
-          return (ssize_t)-ENOMEM;
-        }
-    }
-#endif
-
```

Comment:
Why these were removed?

Reply:
For example reading SPI Flash data, We should copy data from SPI data registers to user buffers, to avoid the buffer address is not aligned by 4 bytes or other case like it is from extram, I firstly copy these data from registers into function's local buffer and then copy data from local buffer into user buffer. This operation maybe a little rough, but it is useful.

Reply:
Did you create smaller critical sections around copying from local buffer to user buffer? (I made a comment somewhere else were I think it's related to this.)

Reply:
Yes, the process that reading data from registers into local buffer and copy data from local buffer into user buffer is in critical sections.